flocker/apiclient/_client.py

```diff
@@ -0,0 +1,90 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Client for the Flocker REST API.
+"""
+
+from uuid import UUID
+
+from zope.interface import Interface
+
+from pyrsistent import PClass, field, pmap_field
+
+
+class Dataset(PClass):
+    """
+    A dataset in the configuration.
+    """
+    dataset_id = field(type=UUID)
+    primary = field(type=UUID)
+    maximum_size = field(type=int)
+    deleted = field(type=bool)
+    metadata = pmap_field(unicode, unicode)
+
+
+class DatasetState(PClass):
+    """
+    The state of a dataset in the cluster.
+    """
+    dataset_id = field(type=UUID)
+    primary = field(type=UUID)
+    maximum_size = field(type=int)
+
+
+class DatasetAlreadyExists(Exception):
+    """
+    The suggested dataset ID already exists.
+    """
+
+
+class IFlockerAPIV1(Interface):
+    """
+    The Flocker REST API, v1.
+    """
+    def create_dataset(primary, maximum_size, dataset_id=None, metadata=None):
+        """
+        Create a new dataset in the configuration.
+
+        :return: ``Deferred`` firing with resulting ``Dataset``, or
+            errbacking with ``DatasetAlreadyExists``.
+        """
+
+    def move_dataset(primary, dataset_id):
+        """
+        Move the dataset to a new location.
+
+        :return: ``Deferred`` firing with resulting ``Dataset``.
```

Comment:
Presumably you'd get some sort of error. But specifying a particular error isn't necessary to proceed with basic docker plugin logic, so not going to bother with that yet.